File: app/src/packages/administrativo/services/PBanco/PBancoShowByCodigoService.ts
'use server';

import { PBancoShowByCodigoData } from '@/packages/administrativo/data/PBanco/PBancoShowByCodigoData';
import { withClientErrorHandler } from '@/shared/actions/withClientErrorHandler/withClientErrorHandler';

async function executePBancoShowByCodigoService(codigoBanco: string) {
  return await PBancoShowByCodigoData(codigoBanco);
}

export const PBancoShowByCodigoService = withClientErrorHandler(executePBancoShowByCodigoService);

File: api/packages/v1/administrativo/services/p_banco/go/p_banco_show_by_codigo_service.py
from fastapi import HTTPException, status

from packages.v1.administrativo.actions.p_banco.p_banco_show_by_codigo_action import (
    ShowByCodigoAction,
)
from packages.v1.administrativo.schemas.p_banco_schema import PBancoCodigoSchema


class ShowByCodigoService:
    def execute(self, codigo_schema: PBancoCodigoSchema):
        data = ShowByCodigoAction().execute(codigo_schema)

        if not data:
            raise HTTPException(
                status_code=status.HTTP_404_NOT_FOUND,
                detail="Não foi possível localizar o banco para o código informado.",
            )

        return data
